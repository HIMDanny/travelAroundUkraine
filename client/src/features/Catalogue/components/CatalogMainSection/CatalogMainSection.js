import React from 'react';
import { styled, Box, Container, Typography } from '@mui/material';

const MainSectionWrapper = styled(Box)(({ theme }) => ({
  backgroundImage: `url(${'./assets/images/cataloguePage/catalogMainSectionBG.jpg'})`,
  backgroundRepeat: 'no-repeat',
  backgroundPosition: 'center',
  backgroundSize: 'cover',
  backgroundColor: theme.palette.primary.main,
  height: 370,
  display: 'flex',
  alignItems: 'center',
 
  [theme.breakpoints.up('tablet')]: {
    height: 510,
  },

  [theme.breakpoints.up('laptop')]: {
    height: 289,
  },
}));

const MainSectionText = styled(Typography)(({ theme }) => ({
  fontWeight: 700,
  fontsize: 36,
  marginBottom: 0,
  color: theme.palette.primary.contrastText,
  textTransform: 'uppercase',
  
  [theme.breakpoints.up('tablet')]: {
    fontsize: 44,
  },

  [theme.breakpoints.up('laptop')]: {
    fontsize: 72,
  },

}));

const CatalogMainSection = () => (
  <MainSectionWrapper>
    <Container>
      <MainSectionText variant='h1'>our tours</MainSectionText>
    </Container>
  </MainSectionWrapper>
);

export default CatalogMainSection;
