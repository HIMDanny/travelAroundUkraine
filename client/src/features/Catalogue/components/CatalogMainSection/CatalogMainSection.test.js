import { render } from '@testing-library/react';
import CatalogMainSection from './CatalogMainSection';

jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Container', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);

describe('CatalogMainSection snapshot testing', () => {
  test('should CatalogMainSection match snapshot', () => {
    const { asFragment } = render(<CatalogMainSection />);

    expect(asFragment()).toMatchSnapshot();
  });
});
