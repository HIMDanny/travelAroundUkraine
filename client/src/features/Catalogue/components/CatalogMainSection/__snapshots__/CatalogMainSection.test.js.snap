// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogMainSection snapshot testing should CatalogMainSection match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <p>
        our tours
      </p>
    </div>
  </div>
</DocumentFragment>
`;
